name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java

```diff
@@ -24,46 +26,92 @@ private ColorSchema(String label)
         this.label = label;
     }
 
-    @Override
-    public String toString()
+    DoubleFunction<Color> buildColorFunction(ResourceManager resourceManager)
     {
-        return label;
+        return switch (this)
+        {
+            case GREEN_YELLOW_RED -> performance -> {
+                // Normalize performance between -0.07 and +0.07 and map it to a
+                // hue between 0 (red) and 120 (green)
+                double p = normalizePerformance(performance);
+
+                // 0 = red, 60 = yellow, 120 = green
+                float hue = (float) p * 120f;
+
+                return resourceManager.createColor(new RGB(hue, 0.9f, 1f));
+            };
+
+            case GREEN_WHITE_RED -> performance -> {
+                // Performance is normalized and interpolated between white
+                // (neutral) and green (positive) or red (negative)
+                double p = Math.min(MAX_PERFORMANCE, Math.abs(performance)) / MAX_PERFORMANCE;
+
+                RGB color = performance > 0f
+                                ? Colors.interpolate(Colors.WHITE.getRGB(), Colors.HEATMAP_DARK_GREEN.getRGB(), (float) p)
+                                : Colors.interpolate(Colors.WHITE.getRGB(), Colors.RED.getRGB(), (float) p);
+
+                return resourceManager.createColor(color);
+            };
+
+            case GREEN_RED_GREY -> performance -> {
+                // Performance interpolates between grey (neutral), green
+                // (positive), or red (negative)
+                double p = normalizePerformance(performance);
```

Comment:
Thinking about it some more, does it really make sense to normalize the performance here?

-0.07 will be normalized to 0, and +0.07 will be normalized to 1.
But then we interpolate depending on the original performance.
So dark gray/green will only be interpolated between 0 and 0.5.
And the gray/red will only be interpolated between 0.5 and 1.